### PR TITLE
Style cleanup for `contrib/browndog/*.R` scripts

### PR DESCRIPTION
**Pull request by bpbond**
Ran through formatR::tidy_dir with arrow=TRUE and indent=2; made sure all if blocks have braces; cracked down on over-length lines; changed `require` calls to `library`.

Note significant (non-style) changes made:
* in `PEcAn#Site_convert.R`, added lines 58-60, providing a failure case for the `if`s
* in various `convert` files, changed `file.path` pattern to "*.nc$"
* use `Sys.which` instead of hard-coded path in `system2` calls

### CODE REVIEW

**Comment by mdietze** [comment]
@robkooper could you review this one too when you get a chance -- there's aspects of Brown Dog syntax to check that you know better than anyone else on the team.

File: contrib/browndog/PEcAn#DALEC_convert.R [hunk 1]
```diff
@@ -14,75 +14,75 @@ verbose <- TRUE
...
     }
-    site <- pieces[1]
-    year <- pieces[2]
-    file.copy(inputFile, file.path(cffolder, paste(site, year, "nc", sep=".")))
-    startDate <- as.POSIXlt(paste0(year,"-01-01 00:00:00"), tz = "GMT")
-    endDate <- as.POSIXlt(paste0(year,"-12-31 23:59:59"), tz = "GMT")
+    if (is.na(endYear) || pieces[2] > endYear) {
+      endYear <- pieces[2]
+    }
+    startDate <- as.POSIXlt(paste0(startYear, "-01-01 00:00:00"), tz = "GMT")
```
Comment (robkooper) [inline contrib/browndog/PEcAn#DALEC_convert.R]: This should be UTC (error in original code). GMT != UTC
Same for the rest of the files.

**Comment by robkooper** [comment]
Can be merged (once updated). @yan130 can do some more cleanup later and fix the issue with GMT.

**Comment by mdietze** [comment]
Just noticed this one had been approved as well. There are some conflicts, but I suspect those are with the recent GMT fix.  Needs and Update and resolve conflicts.  Otherwise ready to go on the pull

File: contrib/browndog/PEcAn#DALEC_convert.R [hunk 2]
```diff
@@ -1,10 +1,10 @@
 #!/usr/bin/Rscript
-#PEcAn
-#data
-#pecan.zip
-#dalec
+# PEcAn
```
Comment (yan130) [inline contrib/browndog/PEcAn#DALEC_convert.R]: @robkooper   I am not sure if this addiction space will break browndog code to parse the input & output file format.

Comment (robkooper) [inline contrib/browndog/PEcAn#DALEC_convert.R]: I'd hope not, if it does we should discuss this with the polyglot people and have them fix it :)

**Pull request merged**